src/python/pants/backend/docker/util_rules/dependencies.py

```diff
@@ -3,30 +3,36 @@
 
 from __future__ import annotations
 
+from dataclasses import dataclass
 from pathlib import PurePath
 
 from pants.backend.docker.subsystems.dockerfile_parser import DockerfileInfo, DockerfileInfoRequest
 from pants.backend.docker.target_types import DockerImageDependenciesField
 from pants.core.goals.package import AllPackageableTargets, OutputPathField
 from pants.engine.addresses import Addresses, UnparsedAddressInputs
 from pants.engine.rules import Get, collect_rules, rule
-from pants.engine.target import InjectDependenciesRequest, InjectedDependencies
+from pants.engine.target import FieldSet, InferDependenciesRequest, InferredDependencies
 from pants.engine.unions import UnionRule
 from pants.util.strutil import softwrap
 
 
-class InjectDockerDependencies(InjectDependenciesRequest):
-    inject_for = DockerImageDependenciesField
+@dataclass(frozen=True)
+class DockerInferenceFieldSet(FieldSet):
+    required_fields = (DockerImageDependenciesField,)
+
+    depenendencies: DockerImageDependenciesField
+
+
+class InferDockerDependencies(InferDependenciesRequest):
+    infer_from = DockerInferenceFieldSet
 
 
 @rule
-async def inject_docker_dependencies(
-    request: InjectDockerDependencies, all_packageable_targets: AllPackageableTargets
-) -> InjectedDependencies:
+async def infer_docker_dependencies(
+    request: InferDockerDependencies, all_packageable_targets: AllPackageableTargets
+) -> InferredDependencies:
```

Comment:
Hooray!

I've been meaning to get to this change for a long time! So happy to see this change 😀